sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -645,6 +645,12 @@ static void rom_ext_rescue_lockdown(boot_data_t *boot_data) {
   flash_ctrl_creator_info_pages_lockdown();
   // Set the OWNER_CONFIG pages for rescue mode (page0=ro, page1=rw).
   ownership_pages_lockdown(boot_data, /*rescue=*/kHardenedBoolTrue);
+  // Lock access to owner-level INFO pages.  During normal boot, this
+  // is performed by `ownership_flash_lockdown`, but we can't call that
+  // function when entering rescue because rescue needs the flash DATA
+  // segments to be writable.  Rescue has no need to access the INFO
+  // pages, so we want to lock them for safety.
+  owner_block_info_lockdown(owner_config.info);
```

Comment:
Can you also add this after `ownership_pages_lockdown()` inside the `rom_ext_boot()` function?

Reply:
`rom_ext_boot` already calls `ownership_flash_lockdown`, which calls `info_lockdown` when the state is `LockedOwner`.

This causes the lock bits to be applied (as requested by the ownership config) in the `LockedOwner` state and causes them to be left unlocked in the various `Unlocked...` states.  The decision to lock or not lock the INFO pages is consistent with the decision to lock or not lock the DATA pages.